### PR TITLE
kubemark: add KUBEMARK_NUM_NODES and KUBEMARK_MASTER_SIZE config

### PR DESCRIPTION
**Pull request by hongchaodeng**
A lot of test infra scripts are using these two parameters and repeatedly set NUM_NODES and MASTER_SIZE before running kubemark. When we try to use those scripts, we need to manually set these again and again.

It would come handy if kubemark config could take these into account and reduce duplication.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37230)

### CODE REVIEW

**Comment by hongchaodeng** [comment]
@gmarek

**Comment by gmarek** [comment]
I'm not sure what bash will do if you default to a variable that's unassgined.

**Comment by hongchaodeng** [comment]
@gmarek 

We are using "{A:-B}" here:
- If A has a value ( len(A) > 0), it will use A
- Otherwise, A=B

If B is unset, it's the same as A is unset. Note that It doesn't matter between "unset" and "empty" here.

File: cluster/kubemark/config-default.sh [hunk 1]
```diff
@@ -24,7 +24,9 @@ source "${KUBE_ROOT}/cluster/gce/config-common.sh"
 GCLOUD=gcloud
 ZONE=${KUBE_GCE_ZONE:-us-central1-b}
 REGION=${ZONE%-*}
+NUM_NODES=${NUM_NODES:-$KUBEMARK_NUM_NODES}
```
Comment (wojtek-t) [inline cluster/kubemark/config-default.sh]: This won't work if KUBEMARK_NUM_NODES is not set. What you should do is:
```
KUBEMARK_NUM_NODES=${KUBEMARK_NUM_NODES:-100}
NUM_NODES=${NUM_NODES:-$KUBEMARK_NUM_NODES}
```
and remove the
```
NUM_NODES=${NUM_NODES:-100}
```
line

**Comment by hongchaodeng** [comment]
@k8s-bot bazel test this

**Comment by hongchaodeng** [comment]
Changed the script a bit. Basically it's needed for KUBMARK_X to take higher priority. PTAL

**Comment by mikedanese** [comment]
@k8s-bot bazel test this

**Comment by mikedanese** [comment]
@k8s-bot bazel test this, issue https://github.com/kubernetes/test-infra/issues/1152

**Comment by mikedanese** [comment]
@k8s-bot bazel test this, issue kubernetes/test-infra#1152

**Comment by gmarek** [comment]
Yeah - my question was if `set -o nounset` won't kill the script in that case.

**Comment by hongchaodeng** [comment]
@gmarek 
That's a good point.
I tested it and it did kill out.
I have fixed it and tested all four cases:
- KUBEMARK_X unset and X unset
- KUBEMARK_X set and X unset
- KUBEMARK_X unset and X set
- KUBEMARK_X set and X set

PTAL

**Comment by wojtek-t** [comment]
lgtm

**Comment by hongchaodeng** [comment]
@wojtek-t 
You need to type "/lgtm" :)

File: cluster/kubemark/config-default.sh [hunk 2]
```diff
@@ -25,7 +25,11 @@ GCLOUD=gcloud
 ZONE=${KUBE_GCE_ZONE:-us-central1-b}
 REGION=${ZONE%-*}
 NUM_NODES=${NUM_NODES:-100}
+# KUBEMARK_NUM_NODES overrides NUM_NODES
+NUM_NODES=${KUBEMARK_NUM_NODES:-$NUM_NODES}
```
Comment (gmarek) [inline cluster/kubemark/config-default.sh]: I'd probably go with `NUM_NODES=${KUBEMARK_NUM_NODES:-${NUM_NODES:-100}}` and the same with `MASTER_SIZE`.

**Comment by hongchaodeng** [comment]
@gmarek Thanks for the suggestion! Fixed

**Comment by hongchaodeng** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by hongchaodeng** [comment]
@gmarek 
How can we get this in?

**Comment by wojtek-t** [comment]
@hongchaodeng - there is complete merge freeze until Monday - it will be merge after that.

**Pull request merged**